config/config.go

```diff
@@ -298,3 +345,142 @@ func (rc *ReporterConfig) NewReporter(
 func (rc *ReporterConfig) newTransport() (jaeger.Transport, error) {
 	return jaeger.NewUDPTransport(rc.LocalAgentHostPort, 0)
 }
+
+// NewTracer returns a new tracer based on the current configuration, using the given options
+func (c *Configuration) NewTracer(options ...Option) (opentracing.Tracer, io.Closer, error) {
+	if c.ServiceName == "" {
+		return nil, nil, errors.New("no service name provided")
+	}
+
+	return c.New(c.ServiceName, options...)
+}
+
+// FromEnv uses environment variables to set the tracer's Configuration
+func FromEnv() *Configuration {
+	c := &Configuration{}
+
+	if e := os.Getenv(envServiceName); e != "" {
+		// We are ignoring any errors, so that we have a single return value
+		// This pattern repeats itself across all the "FromEnv" functions.
+		// TODO: should we at least log when something bad happens, instead of just ignoring?
```

Comment:
I think it's best not to ignore errors. Many env vars are parsed (e.g. as boolean), it's better to fail if we can't parse.

Reply:
This means then that `FromEnv()` has a multiple valued return, making it impossible to do `FromEnv().NewTracer()`, no ?

Reply:
correct, but it's better than swallowing errors.  `FromEnv().NewTracer()` is not exactly the goal here. If people really want a one-liner, we could add `NewTracerFromEnv()`

Reply:
Alright, just wanted to make sure we were on the same page :)